userguide/tutorials/catalog-examples.adoc

```diff
@@ -0,0 +1,380 @@
+= Catalog Examples
+
+This document provides examples of catalogs and how they affect  invoices/subscriptions. 
+
+== How to Use This Document
+
+Each section in this document corresponds to a particular catalog component and provides examples of catalogs that demonstrate that catalog component. 
+
+You can try out the various examples in each section by:
+
+* Uploading the corresponding catalog (via Kaui or https://killbill.github.io/slate/#catalog-upload-a-catalog-as-xml[Upload Catalog] API call)
+* Creating an account (via Kaui or https://killbill.github.io/slate/#account-create-an-account[Create Account] API call)
+
+* Creating a subscription as required (via Kaui or https://killbill.github.io/slate/#subscription-create-a-subscription[Create Subscription] API call)
+* Triggering an invoice dry-run (via Kaui or https://killbill.github.io/slate/#invoice-generate-a-dry-run-invoice[Generate Dry Run] API call)
+
+== Prerequisites
+
+Ensure that you have gone through the https://docs.killbill.io/latest/userguide_subscription.html#components-catalog[Subscription User Guide] and are familiar with the various catalog components like products, plans, phases.
+
+== Billing Modes
+
+A catalog needs to include a *Billing Mode* as explained in the https://docs.killbill.io/latest/invoice_subsystem.html#_overview[invoicing] document. 
+
+Quick summary about Billing modes:
+
+* Can be specified at the *catalog* level or on a *per-plan* basis.
+* *IN_ADVANCE* indicates that each recurring subscription is charged at the _beginning_ of the billing period.
+
+* *IN_ARREAR* indicates that each recurring subscription is charged at the _end_ of the billing period.
+
+This section provides examples of catalogs with different billing modes. For simplicity sake, only catalogs with catalog level billing modes are demonstrated below, however the behavior is the same even when a per-plan billing mode is used.
```

Comment:
@reshmabidikar - Regarding catalog-level versus plan-level billing modes, should we refer them to https://docs.killbill.io/latest/userguide_subscription.html#components-catalog-usage?

Reply:
@FlowWriter, I may be wrong, but I think per-plan billing is not the same as https://docs.killbill.io/latest/userguide_subscription.html#components-catalog-usage. Per-plan billing refers to specifying the billing mode (`IN_ARREAR` or `IN_ADVANCE`) at the plan level instead of the catalog level as explained [here](https://docs.killbill.io/latest/invoice_subsystem.html#_overview) whereas https://docs.killbill.io/latest/userguide_subscription.html#components-catalog-usage refers to usage billing. @pierre , can you please confirm if this is correct?

Reply:
Yup, correct!

_Historical fun fact: we introduced the ability to set the billing mode at the plan level when we first implemented usage-based pricing. Before, only a single global billing mode was supported._